pkg/bytecode/compiler.go

```diff
@@ -13,6 +13,10 @@ var (
 	// ErrUnknownOperator is returned when an operator cannot
 	// be resolved.
 	ErrUnknownOperator = fmt.Errorf("%w: unknown operator", ErrInternal)
+	// ErrUnsupportedExpression is returned when an expression is not
+	// supported by the compiler, this indicates an error in the compiler
+	// itself, as all parseable evy expressions should be supported.
+	ErrUnsupportedExpression = fmt.Errorf("%w: unsupported expression", ErrInternal)
```

Comment:
I guess in theory it could also mean that the parser has made a mistake - either way and internal error.